File: app/src/main/java/com/arctouch/codechallenge/repository/UpcomingMoviesRepository.java
package com.arctouch.codechallenge.repository;

import com.arctouch.codechallenge.entity.UpcomingMoviesResponse;

import io.reactivex.Observable;

public interface UpcomingMoviesRepository {
    Observable<UpcomingMoviesResponse> getUpcomingMovies();
    Observable<UpcomingMoviesResponse> getUpcomingMovies(int page);
}

File: app/src/main/java/com/arctouch/codechallenge/repository/UpcomingMoviesRepository.kt
package com.arctouch.codechallenge.repository

import com.arctouch.codechallenge.entity.UpcomingMoviesResponse

import io.reactivex.Observable

interface UpcomingMoviesRepository {
    val upcomingMovies: Observable<UpcomingMoviesResponse>
    fun getUpcomingMovies(page: Int): Observable<UpcomingMoviesResponse>
}
